internal/controller/rotators/gcp_oidc_token_rotator.go

```diff
@@ -88,6 +87,22 @@ type gcpOIDCTokenRotator struct {
 	stsTokenFunc stsTokenGenerator
 }
 
+// sharedTransport is a shared HTTP transport used for GCP API calls.
+// It is initialized with the GCP proxy URL if provided in the environment variable.
+var (
+	sharedTransport http.RoundTripper
+	initError       error
```

Comment:
Let's rename it to add GCP portion & remove initError
```suggestion
	sharedGCPTransport http.RoundTripper
```